pkg/rtc/mediatrack.go

```diff
@@ -196,13 +197,15 @@ func (t *MediaTrack) AddReceiver(receiver *webrtc.RTPReceiver, track *webrtc.Tra
 	buff, rtcpReader := t.params.BufferFactory.GetBufferPair(uint32(track.SSRC()))
 	if buff == nil || rtcpReader == nil {
 		t.params.Logger.Errorw("could not retrieve buffer pair", nil)
+		prometheus.AddPublishedTrackFailure(t.trackInfo.Type.String())
 		return newCodec
 	}
 
 	rtcpReader.OnPacket(func(bytes []byte) {
 		pkts, err := rtcp.Unmarshal(bytes)
 		if err != nil {
 			t.params.Logger.Errorw("could not unmarshal RTCP", err)
+			prometheus.AddPublishedTrackFailure(t.trackInfo.Type.String())
```

Comment:
this is not the right place for it. this is called on each packet. Also, I think publishing checks can be performed by the supervisor framework.

Another issue with keeping track of failures: we don't always know when something has failed (but will know that they've succeeded). For example, if the onTrack function doesn't fire, we would not be able to account for that.

Therefore, I think in general it's much more predictable to keep track of successes and total. failure rate could be derived from that.